[Executorch][perf-ci] Fix perf ci

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #8374

Summary:
Previous PR #7927 deecoupled max_seq_length from kv cache. That broke
perf ci workflow. Fix that.

Test Plan:
Trigger it manually and check
apple perf: https://github.com/pytorch/executorch/actions/runs/13267110949
android perf: https://github.com/pytorch/executorch/actions/runs/13267110908

Reviewers:

Subscribers:

Tasks:

Tags:

cc @guangy10 @huydhn @kirklandsign @shoumikhin

kimishpatel committed:
[Executorch][perf-ci] Fix perf ci

Summary:
Previous PR #7927 deecoupled max_seq_length from kv cache. That broke
perf ci workflow. Fix that.

Test Plan:
Trigger it manually and check

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

guangy10 commented:
The linked job in the PR summary doesn't run with the SpinQuant and QLora. You need to trigger the job using the model id on Hugging Face:
  - https://huggingface.co/meta-llama/Llama-3.2-1B
  - https://huggingface.co/meta-llama/Llama-3.2-1B-Instruct-SpinQuant_INT4_EO8
  - https://huggingface.co/meta-llama/Llama-3.2-1B-Instruct-QLORA_INT4_EO8

mergennachin commented:
Also the readme pages? 

https://github.com/pytorch/executorch/blob/main/examples/models/llama/README.md

https://github.com/pytorch/executorch/blob/main/examples/demo-apps/apple_ios/LLaMA/docs/delegates/xnnpack_README.md

https://github.com/pytorch/executorch/blob/main/examples/demo-apps/android/LlamaDemo/docs/delegates/xnnpack_README.md

kimishpatel commented:
> doesn't run with the SpinQuant and QLora

let me do this in follow up PR

kimishpatel commented:
> The linked job in the PR summary doesn't run with the SpinQuant and QLora. You need to trigger the job using the model id on Hugging Face:
> 
> * https://huggingface.co/meta-llama/Llama-3.2-1B
> * https://huggingface.co/meta-llama/Llama-3.2-1B-Instruct-SpinQuant_INT4_EO8
> * https://huggingface.co/meta-llama/Llama-3.2-1B-Instruct-QLORA_INT4_EO8

>  need to trigger the job using the model id on Hugging Face:

What does this mean? Is there description as to how to trigger this. I followed steps here https://github.com/pytorch/executorch/tree/main/extension/benchmark

kimishpatel committed:
Update on "[Executorch][perf-ci] Fix perf ci"


Summary:
Previous PR #7927 deecoupled max_seq_length from kv cache. That broke
perf ci workflow. Fix that.

Test Plan:
Trigger it manually and check
apple perf: https://github.com/pytorch/executorch/actions/runs/13267110949
android perf: https://github.com/pytorch/executorch/actions/runs/13267110908

Reviewers:

Subscribers:

Tasks:

Tags:

cc guangy10 huydhn kirklandsign shoumikhin

[ghstack-poisoned]

kimishpatel commented:
> > doesn't run with the SpinQuant and QLora
> 
> let me do this in follow up PR
> 
> Actually let me just do it here

this is updated. But I think I am gonna have to do one more round of scrubbing in subsequent PRs for various incarnations of llama

guangy10 commented:
> > The linked job in the PR summary doesn't run with the SpinQuant and QLora. You need to trigger the job using the model id on Hugging Face:
> > 
> > * https://huggingface.co/meta-llama/Llama-3.2-1B
> > * https://huggingface.co/meta-llama/Llama-3.2-1B-Instruct-SpinQuant_INT4_EO8
> > * https://huggingface.co/meta-llama/Llama-3.2-1B-Instruct-QLORA_INT4_EO8
> 
> > need to trigger the job using the model id on Hugging Face:
> 
> What does this mean? Is there description as to how to trigger this. I followed steps here https://github.com/pytorch/executorch/tree/main/extension/benchmark

You need to specify the models you want to benchmark against explicitly, separated by ",". In this case, they are "meta-llama/Llama-3.2-1B-Instruct-SpinQuant_INT4_EO8,meta-llama/Llama-3.2-1B-Instruct-QLORA_INT4_EO8". See the screenshot for example.

![image](https://github.com/user-attachments/assets/9a00d1e3-cc2e-4335-996f-bfeeae728f02)

guangy10 reviewed (approved):
The changes look good to me

kimishpatel commented:
> > > The linked job in the PR summary doesn't run with the SpinQuant and QLora. You need to trigger the job using the model id on Hugging Face:
> > > 
> > > * https://huggingface.co/meta-llama/Llama-3.2-1B
> > > * https://huggingface.co/meta-llama/Llama-3.2-1B-Instruct-SpinQuant_INT4_EO8
> > > * https://huggingface.co/meta-llama/Llama-3.2-1B-Instruct-QLORA_INT4_EO8
> > 
> > 
> > > need to trigger the job using the model id on Hugging Face:
> > 
> > 
> > What does this mean? Is there description as to how to trigger this. I followed steps here https://github.com/pytorch/executorch/tree/main/extension/benchmark
> 
> You need to specify the models you want to benchmark against explicitly, separated by ",". In this case, they are "meta-llama/Llama-3.2-1B-Instruct-SpinQuant_INT4_EO8,meta-llama/Llama-3.2-1B-Instruct-QLORA_INT4_EO8". See the screenshot for example.
> 
> ![Screenshot 2025-02-11 at 7 30 20 PM](https://private-user-images.githubusercontent.com/42389959/412252745-5c8a7d4c-4b99-47e3-8f5c-71c1ce941b6b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkzMzE2MTUsIm5iZiI6MTczOTMzMTMxNSwicGF0aCI6Ii80MjM4OTk1OS80MTIyNTI3NDUtNWM4YTdkNGMtNGI5OS00N2UzLThmNWMtNzFjMWNlOTQxYjZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjEyVDAzMzUxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI0MjMxYWEzNDFlMGM2MTJkNzQ4YmIwZDUzZWQ5OGY4NGMyZDgzMDY0ZmFjM2UyYmMxMTljNTlkYWFlNDM0N2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wl8lP8Myz73IoXDFahZj_07hhnetiYtX31bhffUCRK4)

oh you are right. I forgot about that step.
